integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -336,11 +340,26 @@ def _get_stream_response(
         :param streaming_callback: The handler for the streaming response.
         :returns: The extracted response with the content of all streaming chunks.
         """
-        responses = []
+        replies: Union[List[str], List[ChatMessage]] = []
+
         for chunk in stream:
-            content = chunk.text if len(chunk.parts) > 0 and "text" in chunk.parts[0] else ""
-            streaming_callback(StreamingChunk(content=content, meta=chunk.to_dict()))
-            responses.append(content)
+            metadata = chunk.to_dict()
+            for candidate in chunk.candidates:
+                for part in candidate.content.parts:
+                    if part.text != "":
+                        replies.append(
+                            ChatMessage(content=part.text, role=ChatRole.ASSISTANT, meta=metadata, name=None)
+                        )
+                    elif part.function_call is not None:
+                        metadata["function_call"] = part.function_call
+                        replies.append(
+                            ChatMessage(
+                                content=dict(part.function_call.args.items()),
+                                role=ChatRole.ASSISTANT,
+                                name=part.function_call.name,
+                                meta=metadata,
+                            )
+                        )
 
-        combined_response = "".join(responses).lstrip()
-        return [ChatMessage.from_assistant(content=combined_response)]
+                streaming_callback(StreamingChunk(content=part.text, meta=chunk.to_dict()))
+        return replies
```

Comment:
Is this ok? Doesn't it change the behaviour of the Component? We were returning a list with a single element with all the responses combined, but not anymore. 🤔

Reply:
This clearly can be improved, do you think the way I did it [here](https://github.com/deepset-ai/haystack-core-integrations/blob/1f53cd944f9a7ef5385039110dc848dc817cf412/integrations/google_vertex/src/haystack_integrations/components/generators/google_vertex/chat/gemini.py#L262) makes sense? This change is also a part of this PR.